hs_core/management/commands/regenerate_xml_files_composite_resource.py

```diff
@@ -0,0 +1,73 @@
+"""Generate bag files composite resources. Regeneration of bag files is needed to get the
+resource level xml files corrected as well as to generate aggregation level
+xml files which currently does not exist for any of the composite resources created prior to
+composite resource phase-2 implementation
+
+* By default, prints errors on stdout.
+* Optional argument --log: logs output to system log.
+"""
+
+from django.core.management.base import BaseCommand
+
+from hs_core.hydroshare import create_bag_files
+from hs_composite_resource.models import CompositeResource
+
+
+class Command(BaseCommand):
+    help = "Generate bag files for specified composite resources"
+
+    def add_arguments(self, parser):
+
+        # a list of resource id's, or none to check all resources
+        parser.add_argument('resource_ids', nargs='*', type=str)
+
+        # Named (optional) arguments
+        parser.add_argument(
+            '--log',
+            action='store_true',  # True for presence, False for absence
+            dest='log',           # value is options['log']
+            help='log errors to system log',
+        )
+
+    def handle(self, *args, **options):
+        resource_counter_success = 0
+        resource_counter = 0
+
+        # inner function
+        def generate_bag_files(resource):
+            metadata_status = {}
+            for aggregation in resource.logical_files:
+                is_dirty = aggregation.metadata.is_dirty
+                metadata_status[aggregation.id] = is_dirty
+                # this is needed to generate aggregation level xml files
+                aggregation.metadata.is_dirty = True
```

Comment:
What do we gain by tracking whether an aggregation metadata is dirty?  It looks like we're just ignoring this during bag creation anyways.

Reply:
Some of the aggregation types (NetCDF and TimeSeries) depend on 'is_dirty' flag to write metadata changes in DB to the respective content files.

Reply:
... ok